### ISSUE TITLE
Find a better lyrics provider 

### ISSUE THREAD

**Issue by filveith** [filveith/BetterDiscord-Spotify-Lyrics-Status#28]
We are currently using api.textyl.co to get the lyrics, with time-stamp, of the song we are currently listening to on spotify.

There are some pros and cons with this API :

**Pros :**
- it's free !
- the lyrics are time-stamped (very important to show the correct lyrics at the right moment)
- the only one available with these two criteria (and that also contains new songs)

**Cons :**
- it's not rare that the api is offline for some hours or even days
- a lot of songs have no lyrics available 
- still not succeeded to contact the supposedly owner of the API

----

To conclude we need an API that is free, has time-stamped lyrics and is updated with new songs

So if you know an API that meets these criteria let us know in this issue ;)

**Comment by BlackLightX** [comment]
"https://genius.com/" can you find the API of this site? Almost all the songs are there.
https://docs.genius.com/

**Comment by BlackLightX** [comment]
> We are currently using api.textyl.co to get the lyrics, with time-stamp, of the song we are currently listening to on spotify.
> 
> There are some pros and cons with this API :
> 
> **Pros :**
> 
> * it's free !
> * the lyrics are time-stamped (very important to show the correct lyrics at the right moment)
> * the only one available with these two criteria (and that also contains new songs)
> 
> **Cons :**
> 
> * it's not rare that the api is offline for some hours or even days
> * a lot of songs have no lyrics available
> * still not succeeded to contact the supposedly owner of the API
> 
> To conclude we need an API that is free, has time-stamped lyrics and is updated with new songs
> 
> So if you know an API that meets these criteria let us know in this issue ;)



"https://genius.com/" can you find the API of this site? Almost all the songs are there.
https://docs.genius.com/

**Comment by filveith** [comment]
Thanks for the proposition, but the problem with genius.com is that they don't have the time-stamp for each lyric. This means that we don't know when we need to display the lyrics in the status.

**Comment by aaashleyyyy** [comment]
i think it is extremely unlikely that your going to find another free API. The only real solution that i can think of is what i said before in a previous issue, somehow figuring out how to grab the lyrics direction from spotify, to avoid using MusixMatch (which is paid) directly. But as you said before, I am not sure of the legality of this, but I also doubt anyone (at MusixMatch or Spotify) would care enough to do anything about it. Figuring out a way to do it in this method would also result in a 100% perfect sync, as oftentimes with the api that is currently being used it is desynced because it's the lyrics of some different version of the song.

**Comment by aaashleyyyy** [comment]
And being able to copy the text from spotify cannot be hard. If you download an app called "textify" you are able to directly copy the text in spotify. So, it's doable.

**Comment by aaashleyyyy** [comment]
https://github.com/SimonIT/spotifylyrics another find that could be useful

**Comment by aaashleyyyy** [comment]
found this tampermonkey script that uses the lyrics directly from spotify

https://github.com/OvalQuilter/lyrics-sender

**Comment by filveith** [comment]
> i think it is extremely unlikely that your going to find another free API. The only real solution that i can think of is what i said before in a previous issue, somehow figuring out how to grab the lyrics direction from spotify, to avoid using MusixMatch (which is paid) directly. But as you said before, I am not sure of the legality of this, but I also doubt anyone (at MusixMatch or Spotify) would care enough to do anything about it. Figuring out a way to do it in this method would also result in a 100% perfect sync, as oftentimes with the api that is currently being used it is desynced because it's the lyrics of some different version of the song.

Yeah you are right, I will try getting the lyrics directly from Spotify. 
Thanks for the links to other repos that have done the same this will help me a lot

**Comment by filveith** [comment]
I will try some things and if they don't work I will deprecate the plugin...

**Comment by filveith** [comment]
We have finally deprecated the plugin, thanks for using it and for your help :))

**Comment by Myst1cX** [comment]
I would like to point towards LRCLIB api.
